### PR TITLE
[FLINK-3580] [table] Implement FLOOR/CEIL for time points

### PR DESCRIPTION
**Pull request by twalthr**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

This PR implements FLOOR/CEIL functions for DATE, TIME, TIMESTAMP types.

### CODE REVIEW

**Comment by twalthr** [comment]
If there are no objections, I would merge this later today.
@fhueske ?

File: docs/apis/table.md [hunk 1]
```diff
@@ -1445,6 +1445,28 @@ TEMPORAL.extract(TIMEINTERVALUNIT)
...
     </tr>
 
+    <tr>
+      <td>
+        {% highlight java %}
+TIMEPOINT.floor(TIMEINTERVALUNIT)
+{% endhighlight %}
+      </td>
+      <td>
+        <p>Rounds down a time point to the given unit. E.g. <code>"12:44:31".toDate.floor(MINUTE)</code> leads to 12:44:00.</p>
```
Comment (fhueske) [inline docs/apis/table.md]: "Rounds a time point down to the given unit." ?

File: docs/apis/table.md [hunk 2]
```diff
@@ -1445,6 +1445,28 @@ TEMPORAL.extract(TIMEINTERVALUNIT)
...
+    </tr>
+
+    <tr>
+      <td>
+        {% highlight java %}
+TIMEPOINT.ceil(TIMEINTERVALUNIT)
+{% endhighlight %}
+      </td>
+      <td>
+        <p>Rounds up a time point to the given unit. E.g. <code>"12:44:31".toTime.floor(MINUTE)</code> leads to 12:45:00.</p>
```
Comment (fhueske) [inline docs/apis/table.md]: "Rounds a time point up to the given unit." ?

**Comment by fhueske** [comment]
Looks good! Thanks, @twalthr.

**Comment by twalthr** [comment]
Thanks @fhueske. I will address your comments and merge this.

**Pull request closed without merging**